### PR TITLE
Add VisualizationTools tutorial

### PR DESCRIPTION
**Pull request by tcNickolas**

### CODE REVIEW

File: tutorials/VisualizationTools/Tests.qs [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    // (see DeutschJozsaAlgorithm kata, task 1.5)
+    operation MarkingOracleSumOfBits (inputRegister : Qubit[], target : Qubit, r : Int[]) : Unit {
+        for ind in r {
+            CNOT(inputRegister[ind], target);
+        }
+    }
+
+
+    // Phase kickback trick.
+    operation ApplyMarkingOracleAsPhaseOracle (markingOracle : ((Qubit[], Qubit) => Unit), inputRegister : Qubit[]) : Unit {
```
Comment (msoeken) [inline tutorials/VisualizationTools/Tests.qs:174]:
```suggestion
    operation ApplyMarkingOracleAsPhaseOracle (markingOracle : (Qubit[], Qubit) => Unit, inputRegister : Qubit[]) : Unit {
```

File: tutorials/VisualizationTools/Tests.qs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    @Test("QuantumSimulator")
+    operation T5_ReadOperationTrace () : Unit {
+        let (expectedQ, expectedH, expectedCNOT) = (4, 8, 2);
+        let (actualQ, actualH, actualCNOT) = ReadMysteryOperationTrace();
+
+        if expectedQ != actualQ or expectedH != actualH or expectedCNOT != actualCNOT {
+            fail "One of the returned values differed from the expected value";
+        }
+    }
+}
```
Comment (msoeken) [inline tutorials/VisualizationTools/Tests.qs:197]:
```suggestion
}

```

**Review by msoeken — APPROVED** [review]
Looks good to me. Thanks! Left some small suggestions.

File: tutorials/VisualizationTools/VisualizationTools.ipynb [hunk 3]
```diff
@@ -0,0 +1,505 @@
...
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Display the circuit performed by the operation\n",
+    "\n",
+    "[Quantum circuits](https://en.wikipedia.org/wiki/Quantum_circuit) are a very common way of visualizing quantum programs; you'll seed them in a lot of tutorials, papers and books on quantum computing.\n",
```
Comment (anjbur) [inline tutorials/VisualizationTools/VisualizationTools.ipynb:386]:
```suggestion
    "[Quantum circuits](https://en.wikipedia.org/wiki/Quantum_circuit) are a very common way of visualizing quantum programs; you'll see them in a lot of tutorials, papers and books on quantum computing.\n",
```

**Review by anjbur — APPROVED** [review]
Found a small typo, but otherwise looks good to me!

**Pull request merged**